File: Corte1/Tarea 120826Entrega140826/SensorMentirosoDinamico.cpp
#include <iostream>
using namespace std;

int main() {
    int cantidadLecturas = 0;
    int i = 0;
    int inputdato = 0;
    int contadorDatos = 0;
    int sumaDatos = 0;
    int contadorMentiras = 0;
    cout << "Ingrese la cantidad de lecturas que desea ingresar: ";
    cin >> cantidadLecturas;

    int lecturas [cantidadLecturas] = {};

    while (i < cantidadLecturas) {
        cout << "Ingrese el dato " << i + 1 << ": ";
        cin >> inputdato;
        lecturas[i] = inputdato;
        i++;
    }
        for (int k = 0; k<cantidadLecturas; k++){
        if (lecturas[k] != -999){
            sumaDatos += lecturas[k];
            contadorDatos++;
        } else {
            contadorMentiras++;
        }
    }
    if (contadorDatos > 0) {
        cout << "El promedio de las lecturas es: " << sumaDatos/contadorDatos << endl;
    }
    cout << "Los datos falsos son: " << contadorMentiras << endl;
    return 0;
}

File: Corte1/Tarea 120826Entrega140826/SensorMentirosoDinamico.py
cantidadDatos = int(input("Ingrese la cantidad de datos que desea ingresar: "))
lecturas = []
contadorMentiras = 0
contadorDatos = 0
sumaDatos = 0

print("Ingrese los datos de las lecturas (ingrese -999 para indicar un dato falso):")

while len(lecturas) < cantidadDatos: 
    inputDato = input(f"Ingrese el dato {len(lecturas) + 1}: ") #El mas 1 es para que se muestre dato 1,2,3 en lugar de 0,1,2
    lecturas.append(int(inputDato))
for i in lecturas:
    if i == -999:
       contadorMentiras += 1
    else:
        contadorDatos += 1
        sumaDatos += i

print(f"Total de datos falsos: {contadorMentiras}")
print(f"Total de datos válidos: {contadorDatos}")
if contadorDatos > 0:
    print(f"Promedio de datos válidos: {sumaDatos / contadorDatos}")